playbooks/cleanup.yaml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Cleanup
+  hosts: all
+  roles:
+  - role: ensure-pip  # https://zuul-ci.org/docs/zuul-jobs/latest/python-roles.html#role-ensure-pip
+  vars:
+    cloud_provider: "{{ cloud }}"  # inherited from the parent job
+    prefix: "pr{{ zuul.change }}"
+    testcluster_name: "pr{{ zuul.change }}"
+  environment:
+    OS_CLOUD: "{{ cloud_provider }}"
+    PATH: "{{ ansible_user_dir }}/.local/bin:{{ ansible_env.PATH }}"
+  tasks:
+  - name: Ensure ospurge and openstackclient
+    pip:
+      name:
+      - "git+https://git.openstack.org/openstack/ospurge"
```

Comment:
Do not use ospurge. It's a dead end. Use openstack purge instead or have a look at https://github.com/osism/testbed/blob/main/terraform/scripts/cleanup.py

Reply:
I tried `openstack project purge` - against *plusserver* gx-scs OpenStack and it doesn't work for me:
```bash
$ openstack project purge --auth-project --keep-project --dry-run
Deleting volume : 333c8620-bbb5-494a-834e-ffbe579a02f9
$ openstack project purge --auth-project --keep-project 
Deleting volume : 333c8620-bbb5-494a-834e-ffbe579a02f9
Failed to delete volume with ID '333c8620-bbb5-494a-834e-ffbe579a02f9': Policy doesn't allow volume_extension:volume_admin_actions:force_delete to be performed. (HTTP 403) (Request-ID: req-85208359-2d7a-4cc7-8ca7-a49ac27197cb)
1 of 1 volumes failed to delete.
```
It discovered only volume and it doesn't even delete it. According to documentation it deletes only `Block Storage v1, v2; Compute v2; Image v1, v2` so maybe that is a problem.

I also tried `neutron purge` - it works but only deletes neutron resources(floating IPs, networks ...).

We will return to this problem of purging resources later (maybe we can create an issue), but for now, I think `ospurge` is sufficient because looks more powerful and it works.